### PR TITLE
Another attempt at fixing the bug with tag_v type

### PR DESCRIPTION
**Pull request by pgnickb**
## Description

Fixes timescale/o11y-team-data-platform#56

We want to redefine the opclass for tag_v type to not use = operator.
This will improve type coercion for the typical use case.
Simply dropping the opclass should be safe since we don't expect anyone
to ever index using this opclass.

NOTE: We currently have two distinct types:

_ps_trace.tag_v and
ps_trace.tag_v
This is not ideal.
## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

**Review by sumerman — COMMENTED** [review]
I think this is good to go, but I'd love another pair of 👁️

File: migration/idempotent/007-tracing-tags.sql [hunk 1]
```diff
@@ -209,20 +210,21 @@ the built-in jsonb. It is the same as its jsonb_ namesake, but relies on tag_map
 
 DO $do$
 BEGIN
-	CREATE OPERATOR ps_trace.<> (
+	DROP OPERATOR IF EXISTS ps_trace.<> (_ps_trace.tag_v, _ps_trace.tag_v)  CASCADE;
```
Comment (JamesGuthrie) [inline migration/idempotent/007-tracing-tags.sql:213]: Is there any reason why these `DROP` statements don't work in an incremental migration? I'm just concerned that we're going to accidentally get something wrong by removing these at some point in the future.

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,9 @@ We use the following categories for changes:
 
 - Added `prom_api.get_default_chunk_interval()` and `prom_api.get_metric_chunk_interval(TEXT)` [#464].
 
+### Fixed
+- Incorrect type coercion when using `tag_map` with `=` operator [#345]
```
Comment (JamesGuthrie) [inline CHANGELOG.md:22]:
```suggestion
- Incorrect type coercion when using `tag_map` with `=` operator [#462]
```

**Review by JamesGuthrie — CHANGES_REQUESTED** [review]

File: migration/idempotent/007-tracing-tags.sql (hunk 1 above)
Comment (pgnickb) [reply migration/idempotent/007-tracing-tags.sql:213]: That's a good question. I was thinking about this as well. Ultimately, I'd prefer related code to be in one place, so it's easier to make changes.
Unfortunately it's hard to achieve that in the current structure of the repo.

Comment (JamesGuthrie) [reply migration/idempotent/007-tracing-tags.sql:213]: I would argue that having the DROP statements in an idempotent script makes it _harder_ to make changes.

Comment (pgnickb) [reply migration/idempotent/007-tracing-tags.sql:213]: I see your point. Let me do some adjustments

**Review by JamesGuthrie — APPROVED** [review]

**Pull request merged**